### ISSUE TITLE
irb3.0 no longer does tab completion on filenames

### ISSUE THREAD

**Issue by simon-dedeo** [ruby/irb#198]
## Description

in irb2.5:

rb(main):001:0> source 'li [then I hit tab...]
linking.rb list_load.rb list_load_OLD.rb

irb(main):001:0> source 'lis [then I hit tab again...]
list_load.rb list_load_OLD.rb

irb(main):001:0> source 'list_load.rb' [then I hit tab again...]

in irb3.0, hitting tab after typing "source 'li" gives.

irb(main):001:0' source 'li'

## Result of irb_info

Ruby version: 3.0.0
IRB version: irb 1.3.4 (2021-02-25)
InputMethod: ReidlineInputMethod with Reline 0.2.0 and /Users/simon/.inputrc
.irbrc path: /Users/simon/.irbrc
RUBY_PLATFORM: x86_64-darwin16

## Terminal Emulator

Mac OS X (10.12.6)

## Setting Files

ayerie:~ simon$ cat .inputrc
"\e[A": history-search-backward
"\e[B": history-search-forward
set show-all-if-ambiguous on
set completion-ignore-case on

ayerie:~ simon$ cat /Users/simon/.irbrc
[standard text, it seems]

```ayerie:~ simon$ cat /Users/simon/.irbrc
module Readline
  module History
    LOG = "#{ENV['HOME']}/.irb-history"

    def self.write_log(line)
      File.open(LOG, 'ab') {|f| f << "#{line}
"}
    end

    def self.start_session_log
      write_log("
# session start: #{Time.now}

")
      at_exit { write_log("
# session stop: #{Time.now}
") }
    end
  end

  alias :old_readline :readline
  def readline(*args)
    ln = old_readline(*args)
    begin
      History.write_log(ln)
    rescue
    end
    ln
  end
end

Readline::History.start_session_log
```

**Comment by aycabta** [comment]
The default completion logic in GNU Readline is filename completion, as IRB-specific completion was disabled before Ruby 2.6, so filenames were completed everywhere. So filename completion no longer works for now.

**Comment by sdedeo** [comment]
Thank you!

**Comment by marcandre** [comment]
Closing, as there won't be a fix.

Or did I misunderstand @aycabta and you are considering doing tab completion for paths when typing a string?

**Comment by aycabta** [comment]
This is implemented by #212.
